examples/niswitch/software-scanning.py

```diff
@@ -123,14 +123,18 @@ def ThrowOnError (vi, errorCode):
         ))).status)
     print("Scanning completed.")
 
-    # Close NI-SWITCH session.
-    CheckForError(vi, (niswitch_client.Close(niswitch_types.CloseRequest(
-        vi = vi
-        ))).status)
-
 # If NI-SWITCH API throws an exception, print the error message  
 except grpc.RpcError as rpc_error:
     error_message = rpc_error.details()
     if rpc_error.code() == grpc.StatusCode.UNAVAILABLE :
         error_message = f"Failed to connect to server on {server_address}"
-    print(f"{error_message}") 
\ No newline at end of file
+    elif rpc_error.code() == grpc.StatusCode.UNIMPLEMENTED:
+        error_message = "The operation is not implemented or is not supported/enabled in this service"
+    print(f"{error_message}") 
+
+finally:
+    if('vi' in vars() and vi.id != 0):
+        # close the session.
+        CheckForError(vi, (niswitch_client.Close(niswitch_types.CloseRequest(
+            vi = vi
+        ))).status)
```

Comment:
In this file, more exception handling ie for UNIMPLEMENTED case is added and also finally-block is added.